src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -662,7 +662,7 @@ PosixEventEngine::CreatePosixEndpointFromFd(int fd,
                                             const EndpointConfig& config,
                                             MemoryAllocator memory_allocator) {
 #if GRPC_PLATFORM_SUPPORTS_POSIX_POLLING
-  GPR_DEBUG_ASSERT(fd > 0);
+  GPR_ASSERT(fd > 0);
```

Comment:
Why change this to a non-debug assert?

Reply:
Now that we have a new use case where `fd` operations are exposed publicly, and since instances of the posix engine can be created and used outside of gRPC, it seems wise to turn this on for all builds.